File: core/src/com/mygdx/game/entity/LandPoint.java
package com.mygdx.game.entity;

import com.badlogic.gdx.maps.MapObject;
import com.mygdx.game.Actor1;

public class LandPoint extends Point {

    public static LandPoint NOTHINIG = new LandPoint(new MapObject(), -1, -1, -1, -1);
    private Actor1 owner;
    private String name = "";
    private int level = 0;
    private int price = 400;
    private int buildPrice = price * 1;

    public String getOwnerName() {
        if (null == owner) {
            return "";
        }
        return owner.getName();
    }

    public Actor1 getOwner() {
        return owner;
    }

    public void setOwner(Actor1 owner) {
        this.owner = owner;
    }

    public LandPoint(MapObject object, int row, int col, int x, int y) {
        super(object, row, col, x, y);
    }

    public boolean isNothing() {
        return NOTHINIG.equals(this);
    }

    public boolean isBlank() {
        return "".equals(getOwnerName());
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getPrice() {
        return price;
    }

    public void setPrice(int price) {
        this.price = price;
    }

    public int getBuildPrice() {
        return price * level;
    }

    public void setBuildPrice(int buildPrice) {
        this.buildPrice = buildPrice;
    }

    public void buildUp() {
        levelUp();
        setPrice(getPrice() * 2);
    }

    public void levelUp() {
        if (level < 5) {
            level++;
        }
    }

    public void levelDown() {
        if (level > 0) {
            level--;
        }
    }

    public int getLevel() {
        return level;
    }

    public boolean isCanLevelUp() {
        return getLevel() < 5;
    }

    public String getBuildingTiledName() {
        return name + "_" + level;
    }

}

File: core/src/main/kotlin/com/rich/diy/game/entity/LandPoint.kt
package com.rich.diy.game.entity

import com.badlogic.gdx.maps.MapObject
import com.rich.diy.game.Actor1

class LandPoint(`object`: MapObject, row: Int, col: Int, x: Int, y: Int) : Point(`object`, row, col, x, y) {
    private var owner: Actor1? = null
    var name = ""
    var level = 0
        private set
    var price = 400
    var buildPrice = price * 1
        get() = price * level
    val ownerName: String
        get() = if (null == owner) {
            ""
        } else owner!!.getName()

    fun getOwner(): Actor1? {
        return owner
    }

    fun setOwner(owner: Actor1?) {
        this.owner = owner
    }

    val isNothing: Boolean
        get() = NOTHINIG == this
    val isBlank: Boolean
        get() = "" == ownerName

    fun buildUp() {
        levelUp()
        price = price * 2
    }

    fun levelUp() {
        if (level < 5) {
            level++
        }
    }

    fun levelDown() {
        if (level > 0) {
            level--
        }
    }

    val isCanLevelUp: Boolean
        get() = level < 5
    val buildingTiledName: String
        get() = name + "_" + level

    companion object {
        var NOTHINIG = LandPoint(MapObject(), -1, -1, -1, -1)
    }
}